news/12175.removal.rst

```diff
@@ -0,0 +1 @@
+Drop a fallback to using SecureTransport on macOS as it is not needed anymore.
```

Comment:
Could you put further information (already present in PR description) here? This text appears in the release note and it would be nice if users can have the context in the notes. For example, if they already have OpenSSL 1.1 they can be relatively confident this does not affect them at all.

Reply:
Good idea, thanks! I updated the news entry, please let me know if you have any suggestions for the wording